neonvm/apis/neonvm/v1/virtualmachine_types.go

```diff
@@ -150,6 +158,11 @@ type VirtualMachineSpec struct {
 	// propagated to the VM.
 	// +optional
 	TargetRevision *RevisionWithTime `json:"targetRevision,omitempty"`
+
+	// Controls how CPU scaling is performed, either hotplug new CPUs with QMP, or enable them in sysfs.
+	// +kubebuilder:validation:Enum=qmpScaling;sysfsScaling
+	// +optional
+	CpuScalingMode *string `json:"cpuScalingMode,omitempty"`
```

Comment:
suggestion: I would rather use a distinct type here, instead of `string`. IIRC you can then do the `kubebuilder:validation:Enum` on the type definition (see e.g. `RestartPolicy` in this file)

(with that, a small nit: the values should probably be upper camel case)

Reply:
changed it to be type + used kubebuilder:validation:Enum at the type level

Reply:
nice, looks good!